experimental/AffinePlaneCurve.jl

```diff
@@ -0,0 +1,355 @@
+  B = eliminate(I, [gen(R, 1)])
+  # We factorize the polynomial we obtain (which is a polynomial where only
+  # the second variable appears).
+  Z = factor(gen(B, 1))
+  Y = []
+  L = Array{S, 1}[]
+  # For the linear factors of g, we add the constant coefficient to the list Y.
+  # The constant coefficient is stored as a mpoly to be used later in evaluate.
+  for g in keys(Z.fac)
+     if total_degree(g) == 1
+        f = g//lc(g)
+        push!(Y, -f + gen(R, 2))
+     end
+  end
+  if !isempty(Y)
+     # For each y, we compute the possible values of x by replacing the second
+     # variable by the value y, and factorizing the resulting polynomial.
+     for y in Y
+        FF = evaluate(F, [gen(R, 1), y])
+        HH = evaluate(H, [gen(R, 1), y])
+        GG = gcd(FF, HH)
+        ZZ = factor(GG)
+        for g in keys(ZZ.fac)
+           if total_degree(g) == 1
+              f = g//lc(g)
+              # We use lc to convert the mpoly into element of S.
+              push!(L, [lc(-f + gen(R, 1)), lc(y)])
+           end
+        end
+     end
+  end
+  # L contains the intersection points as arrays (or can be empty).
+  Pts = Array{Point, 1}()
+  CC = Array{AffinePlaneCurve, 1}()
+  if !isempty(L)
+     for p in L
+        push!(Pts, Point(p))
+     end
+  end
+  if !isone(G)
+     push!(CC, AffinePlaneCurve(G))
+  end
+  return [CC, Pts]
+end
+
+################################################################################
+# Intersection in the sense of varieties.
+# Might change depending on future changes on VarietyModule.
+
+@doc Markdown.doc"""
+    intersect( C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+
+Return the variety defined by the intersection of `C` and `D`.
+"""
+function Oscar.intersect(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+  I = ideal(parent(C.eq), [C.eq, D.eq])
+  V = Variety(I)
+  return V
+end
+
+################################################################################
+# Compute the singular locus of an affine plane curve by intersecting the
+# derivatives and then checking if the intersection belongs also to the
+# curve. The points might also be contained in the components.
+
+@doc Markdown.doc"""
+    curve_singular_locus(C::AffinePlaneCurve)
+
+Return the reduced singular locus of `C` as a list whose first element is the affine plane curve consisting of the singular components of `C` (if any), and the second element is the list of the isolated singular points (which may be contained in the singular component). The singular component might not contain any point over the considered field.
+"""
+function curve_singular_locus(C::AffinePlaneCurve)
+   comp = curve_components(C)
+   D = reduction(C)
+   Pts = Array{Point, 1}()
+   CC = Array{AffinePlaneCurve, 1}()
+   # The components with multiplicity > 1 are singular
+   f = []
+   for (h, c) in comp
+      if c != 1
+         push!(f, h.eq)
+      end
+   end
```

Comment:
This might also work (untested)
```suggestion
   f = [h.eq for (h,c) in comp if c != 1]
```